scipy/optimize/minpack.py

```diff
@@ -449,23 +448,37 @@ def leastsq(func, x0, args=(), Dfun=None, full_output=0,
         return (retval[0], info)
 
 
-def _wrap_func(func, xdata, ydata, weights):
-    if weights is None:
+def _wrap_func(func, xdata, ydata, T):
```

Comment:
Surhud More a écrit :
> In scipy/optimize/minpack.py
> <https://github.com/scipy/scipy/pull/6493#discussion_r75631341>:
>
>> @@ -449,23 +448,37 @@ def leastsq(func, x0, args=(), Dfun=None, full_output=0,
>>          return (retval[0], info)
>>
>>
>> -def _wrap_func(func, xdata, ydata, weights):
>> -    if weights is None:
>> +def _wrap_func(func, xdata, ydata, T):
>
> @dlax <https://github.com/dlax> Thanks for going through this. @nmayorov
> <https://github.com/nmayorov> came up with T for "transformation",
> basically depending upon whether you pass errors or covariance, T
> performs a transformation on the residuals to be optimized. Do you think
> adding some comments at the top of these functions will help?
>

The parameter name could then be more explicit and a comment wouldn't 
hurt. I was suggesting that this transformation be handled by a 
dedicated function that'd wrap the result of `_wrap_func`, but a better 
name and a comment might be enough.